### PR TITLE
Refactor: support entire page OCR with `ocr_mode` and `ocr_languages`

### PR DESCRIPTION
**Pull request by yuming-long**
## Summary
Second part of OCR refactor to move it from inference repo to unstructured repo, first part is done in https://github.com/Unstructured-IO/unstructured-inference/pull/231. This PR adds OCR process logics to entire page OCR, and support two OCR modes, "entire_page" or "individual_blocks".

The updated workflow for `Hi_res` partition:
* pass the document as data/filename to inference repo to get `inferred_layout` (DocumentLayout)
* pass the document as data/filename to OCR module, which first open the document (create temp file/dir as needed), and split the document by pages (convert PDF pages to image pages for PDF file)
* if ocr mode is `"entire_page"`
    *  OCR the entire image
    * merge the OCR layout with inferred page layout
 * if ocr mode is `"individual_blocks"`
    *  from inferred page layout, find element with no extracted text, crop the entire image by the bboxes of the element
    * replace empty text element with the text obtained from OCR the cropped image
* return all merged PageLayouts and form a DocumentLayout subject for later on process

This PR also bump `unstructured-inference==0.7.2` since the branch relay on OCR refactor from unstructured-inference.
  
## Test
```
from unstructured.partition.auto import partition

entrie_page_ocr_mode_elements = partition(filename="example-docs/english-and-korean.png", ocr_mode="entire_page", ocr_languages="eng+kor", strategy="hi_res")
individual_blocks_ocr_mode_elements = partition(filename="example-docs/english-and-korean.png", ocr_mode="individual_blocks", ocr_languages="eng+kor", strategy="hi_res")
print([el.text for el in entrie_page_ocr_mode_elements])
print([el.text for el in individual_blocks_ocr_mode_elements])
```
latest output:
```
# entrie_page
['RULES AND INSTRUCTIONS 1. Template for day 1 (korean) , for day 2 (English) for day 3 both English and korean. 2. Use all your accounts. use different emails to send. Its better to have many email', 'accounts.', 'Note: Remember to write your own "OPENING MESSAGE" before you copy and paste the template. please always include [TREASURE HARUTO] for example:', '안녕하세요, 저 희 는 YGEAS 그룹 TREASUREWH HARUTOM|2] 팬 입니다. 팬 으 로서, HARUTO 씨 받 는 대 우 에 대해 의 구 심 과 불 공 평 함 을 LRU, 이 일 을 통해 저 희 의 의 혹 을 전 달 하여 귀 사 의 진지한 민 과 적극적인 답 변 을 받을 수 있 기 를 바랍니다.', '3. CC Harutonations@gmail.com so we can keep track of how many emails were', 'successfully sent', '4. Use the hashtag of Haruto on your tweet to show that vou have sent vour email]', '메 고']
# individual_blocks
['RULES AND INSTRUCTIONS 1. Template for day 1 (korean) , for day 2 (English) for day 3 both English and korean. 2. Use all your accounts. use different emails to send. Its better to have many email', 'Note: Remember to write your own "OPENING MESSAGE" before you copy and paste the template. please always include [TREASURE HARUTO] for example:', '안녕하세요, 저 희 는 YGEAS 그룹 TREASURES HARUTOM| 2] 팬 입니다. 팬 으로서, HARUTO 씨 받 는 대 우 에 대해 의 구 심 과 habe ERO, 이 머 일 을 적극 저 희 의 ASS 전 달 하여 귀 사 의 진지한 고 2 있 기 를 바랍니다.', '3. CC Harutonations@gmail.com so we can keep track of how many emails were ciiccecefisliy cant', 'VULLESSIULY Set 4. Use the hashtag of Haruto on your tweet to show that you have sent your email']
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,11 @@ install-base: install-base-pip-packages install-nltk-models
 install: install-base-pip-packages install-dev install-nltk-models install-test install-huggingface install-all-docs
 
 .PHONY: install-ci
-install-ci: install-base-pip-packages install-nltk-models install-huggingface install-all-docs install-test
+install-ci: install-base-pip-packages install-nltk-models install-huggingface install-all-docs install-test install-local-inference-branch
+
+.PHONY: install-local-inference-branch
+install-local-inference-branch:
+	git clone -b  yuming/remove_ocr_code --single-branch https://github.com/Unstructured-IO/unstructured-inference.git && cd unstructured-inference && pip install -e . && cd ../
```
Comment (yuming-long) [inline Makefile:28]: this change is for review ci tests only, since all the tests need to use the updated inference repo with ocr related code removed

**Review by benjats07 — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (yuming-long) [reply Makefile:28]: also the same purpose for all changes related to CI files, just to make sure CI use the inference refactor branch. will be removed after the inference refactor branch is merged and publish a new inference version...

**Review by qued — CHANGES_REQUESTED** [review]
Before we merge this... ideally this refactor should not produce any changes to the ingest tests as we are trying to reproduce the process that normally occurs in `unstructured-inference` here instead. I see there are ingest test differences, and that might be fine, but can you give a description of why there are differences, and why we are okay with them?

File: unstructured/partition/ocr.py [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+            inferred_page_layout.elements,
+            ocr_layout,
+        )
+        inferred_page_layout.elements[:] = merged_page_layout_elements
+        return inferred_page_layout
+    elif ocr_mode == OCRMode.INDIVIDUAL_BLOCKS.value:
+        elements = inferred_page_layout.elements
+        for i, element in enumerate(elements):
+            if element.text == "":
+                cropped_image = image.crop((element.x1, element.y1, element.x2, element.y2))
```
Comment (christinestraub) [inline unstructured/partition/ocr.py:189]: You'll need to implement the same padding logic as in the `unstructured inference` repo - 
https://github.com/Unstructured-IO/unstructured-inference/blob/main/unstructured_inference/inference/elements.py#L270-L271

**Review by christinestraub — CHANGES_REQUESTED** [review]

File: unstructured/partition/ocr.py (hunk 2 above)
Comment (yuming-long) [reply unstructured/partition/ocr.py:190]: thanks so much for catching this! this fixed a unit test after i merge from main that change default model to yolox

**Review by cragwolfe — APPROVED** [review]

**Review by christinestraub — APPROVED** [review]

**Pull request merged**